pkg/volume/csi/csi_mounter.go

```diff
@@ -281,16 +281,6 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	}
 
 	err = saveVolumeData(parentDir, volDataFileName, volData)
-	defer func() {
```

Comment:
> I think, we need to rethink when it is safe to remove CSI json file and when it isn't. I think if NodePublish succeeded or failed with uncertain error then - csi related json file shouldn't be removed. 

I removed this code block. the volume state file can not be removed only when NodePublish is return an finished error.